crates/accelerate/src/quantum_circuit/py_ext.rs

```diff
@@ -0,0 +1,45 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2023
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+//! Contains helper functions for creating [Py<T>] (GIL-independent)
+//! objects without creating an intermediate owned reference. These functions
+//! are faster than PyO3's list and tuple factory methods when the caller
+//! doesn't need to dereference the newly constructed object (i.e. if the
+//! resulting [Py<T>] will simply be stored in a Rust struct).
+//!
+//! The reason this is faster is because PyO3 tracks owned references and
+//! will perform deallocation when the active [GILPool] goes out of scope.
+//! If we don't need to dereference the [Py<T>], then we can skip the
+//! tracking and deallocation.
+
```

Comment:
It hardly matters at all, since we don't actually build Rust documentation other than the parts that PyO3 exposes via Python space, but in theory should these lines be `///!` rather than `//!`?  `//` is just a regular comment, so the `!` wouldn't affect it, I think?

Reply:
There are two kinds of doc comments in Rust, `///` and `//!`. The first one attaches to the item that follows it, and the second to the _parent_. In this case, that's the `py_ext` crate 🙂.

Reply:
Ah, I knew that, I was just mistaken about how many slashes were needed. I was thinking about the `!` as modifying an existing token (like `#!` vs `#` in proc-macro attributes), so I thought the parent doc comment was `///!` but that's mistaken.